dev/TeachingTip/TeachingTip.cpp

```diff
@@ -971,27 +975,58 @@ void TeachingTip::OnF6AcceleratorKeyClicked(const winrt::CoreDispatcher&, const
             m_hasF6BeenInvoked = true;
         }
 
-        winrt::Button f6Button = [this]() -> winrt::Button
+        auto const focusFromContent = [this, args]()
         {
-            auto firstButton = m_closeButton.get();
-            auto secondButton = m_alternateCloseButton.get();
-            if (CloseButtonContent())
+            auto parent = winrt::FocusManager::GetFocusedElement();
+            auto const rootElement = m_rootElement.get();
+            while (parent)
             {
-                std::swap(firstButton, secondButton);
+                if (auto parentAsUIElement = parent.try_as<winrt::UIElement>())
+                {
+                    if (parentAsUIElement == rootElement)
+                    {
+                        return true;
+                    }
+                }
+                parent = winrt::VisualTreeHelper::GetParent(parent.try_as<winrt::DependencyObject>());
             }
-            if (firstButton && firstButton.Visibility() == winrt::Visibility::Visible)
+            return false;
+        }();
+
+        if (focusFromContent)
+        {
+            if (auto&& previouslyFocusedElement = m_previouslyFocusedElement.get())
```

Comment:
>m_previouslyFocusedElement [](start = 50, length = 26)

Should m_previouslyFocusedElement be cleared after this focus call?

Reply:
Sure, that seems reasonable. I'm also going to be cleaning up this logic in an iteration I'm about to push.

---
In reply to: [270931712](https://github.com/Microsoft/microsoft-ui-xaml/pull/511#discussion_r270931712) [](ancestors = 270931712)